File: 3-leetcode/01-easy/27-remove-element.ts
const removeElement = (nums: number[], val: number): number => {
  const length = nums.length;
  let index: number = 0;

  for (let i = 0; i < length; i++) {
    if (nums[i] != val) {
      nums[index] = nums[i];
      index++;
    }
  }

  return index;
};

File: 3-leetcode/01-easy/cpp/27-remove-element.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    int removeElement(vector<int> &nums, int val)
    {
        int index = 0;
        int length = nums.size();
        for (int i = 0; i < length; i++)
        {
            if (nums[i] != val)
            {
                nums[index] = nums[i];
                index++;
            }
        }
        return index;
    }
};